dev/release/post-08-remove-old-artifacts.sh

```diff
@@ -43,7 +43,7 @@ old_releases=$(
   grep -E '^arrow-[0-9]' | \
   sort --version-sort --reverse | \
   tail -n +2 | \
-  tac
+  sort --version-sort
```

Comment:
is the sort even relevant at this point? we are already filtering the last release and we have a list of releases to remove.

Reply:
I don't see how it could be, I left it in just in case there was a reason `tac` was added in the first place.

Reply:
Oh, sorry. I didn't know that `tac` doesn't exist on macOS.

Let's remove this entirely. It's just for removing old releases from older to newer.

Reply:
sounds good. Removed in https://github.com/apache/arrow/pull/47250/commits/1a735e96c2ef76227c65f6a0e59e19a231fd2acb.